cloner/cmd/clone.go

```diff
@@ -78,20 +84,44 @@ func clone(cmd *cobra.Command, args []string) {
 
 	endpoint, err := transport.NewEndpoint(remote)
 	if err != nil {
-		log.Fatal("failed to parse remote", err)
+		log.Fatal("failed to parse remote: ", err)
 	}
 
 	splittedRepo := strings.FieldsFunc(endpoint.Path, func(c rune) bool { return c == '/' }) // FieldsFunc handles repeated and beginning/ending separator characters more sanely than Split
-	if len(splittedRepo) < 2 {
-		log.Fatal("expecting <user>/<repo> in url path, received: ", endpoint.Path)
+	if !(len(splittedRepo) > 0) {
+		log.Fatal("expecting repo in url path, received: ", endpoint.Path)
 	}
 	projectName := splittedRepo[len(splittedRepo)-1]
+	projectName = strings.TrimSuffix(projectName, ".git")
 
 	clonePath := projectName
 	if path != "" {
 		clonePath = path + "/" + projectName
 	}
 
+	if !strategy.Equal("default") {
+		_, err := os.Stat(clonePath)
+
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			log.Fatal("unexpected error: ", err)
+		}
+
+		if err == nil {
```

Comment:
[I am out of my turf, but I am interested nonetheless]
This check is not necessary no? If `err != nil` we already exited with RC 1 in the block above. For readability I would rather remove this `if`.

Reply:
Right, sorry I got confused (and that is my point): too many nesting levels and negations for me to follow.

I think that I would rewrite this block as following:
```go
_, err := os.Stat(clonePath)

if err != nil && !errors.Is(err, os.ErrNotExist) {
  log.Fatal("unexpected error: ", err)
}

if errors.Is(err, os.ErrNotExist) {
  clonePathExists := true
}

if preCloningStrategy.Equal(NotIfExist) && clonePathExists {
  log.Print(clonePath, " already exist, doing nothing")
  os.Exit(0)
}

if preCloningStrategy.Equal(Overwrite)  && clonePathExists {
  log.Print("deleting clone")
  err = os.RemoveAll(clonePath + "/")
  if err != nil {
	log.Fatal("failed to remove existing clone: ", err)
  }
}
 
# Bonus log message, just for clarity
if preCloningStrategy.Equal(NoStrategy)  && clonePathExists {
 log.Print(clonePath, " already exists, but no pre-cloning strategy selected: git will handle this.")
}
```

Would you find this more readable? I would prefer it, but I will let you decide.

Reply:
Your code is wrong on several places:
```
if errors.Is(err, os.ErrNotExist) {
  clonePathExists := true
}
```
should rather be: 
```
if errors.Is(err, os.ErrNotExist) {
  clonePathExists := false
}
```
but even so, clonePathExists lifetime is only within this if block so you will have two errors:
- Unused variable within that if block
- Undefined variable outside that if block

For the bonus log message, whether clonePathExists is true or false does not matter, the goal of this one is to just do nothing.

Anyway, your suggestion gave me a new idea on the implementation that I think should be satisfactory.

Reply:
Yes, sorry the code was sort of a braindump. But I see that you understood what I meant and I find the new implementation much much easier to follow!
Thank you!